docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+
+```yaml
+location:
+   name: tk-multi-setframerange
+   type: app_store
+   version: v0.1.7
+```
+This allows Toolkit (for example the tank updates command) to check when updates are available, update and maintain configurations in a very safe way. 
+Whenever the updates command is run and a new version is available, Toolkit will download the code and place it in a local "bundle cache" on disk and ensure that users have access to it.
+
+There are a few different options for sourcing your app releases.
+
+- [Git](https://developer.shotgunsoftware.com/tk-core/descriptor.html#tracking-against-tags-in-git) and [GitHub](https://developer.shotgunsoftware.com/tk-core/descriptor.html#tracking-against-releases-on-github)
+- [Shotgun Uploads](https://developer.shotgunsoftware.com/tk-core/descriptor.html#pointing-at-a-file-attachment-in-shotgun)
+- [Local paths](https://developer.shotgunsoftware.com/tk-core/descriptor.html#pointing-to-a-path-on-disk)
+
+In your production config, you add your app and switch to using the descriptor that suits your needs.
+
+### Git-based descriptors
+
+We recommend that you start using source control to make sure that changes are tracked, regardless of if you intend to use a git descriptor.
+
+Toolkit supports git (http://git-scm.com/) out of the box, and when running `tank updates` on a configuration, it can check for the latest release in your git repository and update your configuration, just like with the app store descriptor.
+
+The requirements for this are:
+
+- Your git repo needs to contain just a single app
+- Your git repo should have the same structure as the [starter app repository](https://github.com/shotgunsoftware/tk-multi-starterapp).
+- You use [Semantic Versioning](http://semver.org) when creating your tags. 
+Toolkit will use these version numbers to try to determine which version is the most recent, and by following the convention `vX.Y.Z`.
+
+Once you have created your first tag in git (eg. `v1.0.0`), you could then set up your config to use a git descriptor that points to your tag.
+Then you can simply run `tank updates`, and if new tags have been created, you will be prompted if you want to upgrade. 
+The workflow is now identical to the one which happens with official app store apps.
+
+{% include warning title="Caution" content="The git descriptor works well with [centralized configs](https://developer.shotgunsoftware.com/tk-core/initializing.html#centralized-configurations), where the caching of apps is usually run by an admin and is stored to a central location where all user's can access it. However, if you are using a [distributed config](https://developer.shotgunsoftware.com/tk-core/initializing.html#distributed-configurations), then it may not be as suitable. Your app will be downloaded per user, which means each user will need to have git installed and setup to authenticate with your repo, and access the code." %}
```

Comment:
-  no apostrophe in "...where all *users* can access it"
- i'd omit the comma after "authenticate with your repo"